spyder/utils/programs.py

```diff
@@ -462,6 +462,24 @@ def is_module_installed(module_name, version=None, installed_version=None,
             
             return check_version(actver, version, symb)
 
+def is_python_interpreter(filename):
+    """Evaluate wether a file is a python interpreter or not."""
+    real_filename = os.path.realpath(filename) # To follow symlink if existent
+    if not osp.isfile(real_filename) or encoding.is_text_file(real_filename):
+        return False
+    possible_names = ['python', 'python2', 'python2.7', 'python3', 'python3.4',
+                      'python3.5']

```

Comment:
Python 3.6 is our, should we have a more robust thing moving forward?, otherwise we will need to update this every now and then (which seems something we can easily forget)